### PR TITLE
Add new defaults for the icon and label of the phoenix menu item

### LINKED ISSUE #227: Evaluate Switch between classic frontend and ocis-web
- Add a switch between frontends using navigation items (app launcher)
- Improve wording and icons for the switches
- Links (private/public) should redirect to the ocis-web UI

## Documentation
**-> decide where to put it**

### Motivation / abstract
We want to let users switch between classic and new design as easy as possible. For this both classic and new frontend need their own URL (can be relative urls).

### Configure oc10
As soon as phoenix is configured in oc10, a menu item becomes available in the app switcher. By default it shows an icon with a browser window and arrows to the left and right and `New Design` as label. Both the icon and the label can be configured by setting `phoenix.icon` (absolute or relative URL to an svg file) and `phoenix.label` (text of your choice) respectively.

### Configure phoenix
By default there is no link back to the oc10 frontend. However, you can add an application item in your `config.json` file like this:
```
  "applications": [
          {
                  "title": {
                          "en": "Classic Design",
                  },
                  "icon": "switch_ui",
                  "url": "https://url-of-your-oc10-instance",
                  "target": "_self"
          }
  ]
```

- The icon can be picked from [ownCloud design system icons](https://owncloud.design/#/Design%20Tokens/Icon) by name or as an absolute URL.
- the URL needs to point to your ownCloud 10 instance
- If you don't provide `_self` as a target (for opening the url in the same tab/window), it will be opened in a new tab/window by default.

You can add more items to the applications list (e.g. directly to any oc10 app) if needed. Native phoenix apps will be first in the app switcher, while items from the `applications` list will be the last items in the app switcher.

### PR DESCRIPTION
**Pull request by kulmann**
## Description
When phoenix is configured (phoenix.baseUrl in config.php), the app switcher has a menu item for Phoenix. This now has a new and configurable default icon and default label, as discussed and decided by the design initiative last week.

Open question: Is it relevant to have translations for the menu item label?

## Related Issue
- Fixes https://github.com/owncloud/product/issues/227

## Motivation and Context
Make it easier for users to spot the switch to the new design.

## How Has This Been Tested?
✋ 

## Screenshots (if appropriate):
Screenshot 2020-09-21 at 14 29 11
Screenshot 2020-09-21 at 14 29 24

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by pmaier1** [comment]
Nice work, thanks!

> Open question: Is it relevant to have translations for the menu item label?

Yes, please!

File: lib/private/NavigationManager.php [hunk 1]
```diff
@@ -193,11 +193,13 @@ private function init() {
...
 		if ($phoenixBaseUrl) {
-			$iconPath = $this->urlGenerator->imagePath('core', 'default-app-icon.svg');
+			$iconPath = $this->config->getSystemValue('phoenix.icon', $this->urlGenerator->imagePath('core', 'apps/phoenix.svg'));
+			$label = $this->config->getSystemValue('phoenix.label', 'New Design');
+			$l = $this->l10nFac->get("core");
 			$this->add([
 				'id' => 'phoenix',
 				'href' => $phoenixBaseUrl,
-				'name' => 'Phoenix',
+				'name' => $l->t($label),
```
Comment (DeepDiver1975) [inline lib/private/NavigationManager.php:202]: no need to translate this - there is no string that can be extracted and translate by anybody.
If you want 'New Design' to be translated - call $l->t() in line 197

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

File: lib/private/NavigationManager.php (hunk 1 above)
Comment (kulmann) [reply lib/private/NavigationManager.php]: Fixed, thank you!

**Review by micbar — APPROVED** [review]

**Comment by kulmann** [comment]
@micbar codecov results are missing...

**Comment by phil-davis** [comment]
codecov results are almost always missing in core. I had a support ticket about that - https://codecov.freshdesk.com/support/tickets/3634 - but no joy yet. Some days the results get posted back to GitHub, some days not.

**Comment by kulmann** [comment]
Ah, ok. Thanks for explaining @phil-davis :-)

**Pull request merged**